CIScripts/BuildFunctions.ps1

```diff
@@ -0,0 +1,151 @@
+class Repo {
+    [string] $Url;
+    [string] $Branch;
+    [string] $DefaultBranch;
+    [string] $Dir;
+
+    Repo ([string] $Url, [string] $Branch, [string] $Dir, [string] $DefaultBranch) {
+        $this.Url = $Url
+        $this.Branch = $Branch
+        $this.Dir = $Dir
+        $this.DefaultBranch = $DefaultBranch
+    }
+}
+
+function Clone-Repos {
+    Param ([Parameter(Mandatory = $true, HelpMessage = "List of repos to clone")] [Repo[]] $Repos)
+    
+    Write-Host "Cloning repositories"
+    $CustomBranches = @($Repos.Where({ $_.Branch -ne $_.DefaultBranch }) | Select-Object -ExpandProperty Branch -Unique)
+    $Repos.ForEach({
+        # If there is only one unique custom branch provided, at first try to use it for all repos.
+        # Otherwise, use branch specific for this repo.
+        $CustomMultiBranch = $(if ($CustomBranches.Count -eq 1) { $CustomBranches[0] } else { $_.Branch })
+
+        Write-Host $("Cloning " +  $_.Url + " from branch: " + $CustomMultiBranch)
+        git clone -b $CustomMultiBranch $_.Url $_.Dir
+
+        if ($LASTEXITCODE -ne 0) {
+            Write-Host $("Cloning " +  $_.Url + " from branch: " + $_.Branch)
+            git clone -b $_.Branch $_.Url $_.Dir
+
+            if ($LASTEXITCODE -ne 0) {
+                throw "Cloning from " + $_.Url + " failed"
+            }
+        }
+    })
+}
+
+function Contrail-Common-Actions {
+    Param ([Parameter(Mandatory = $true)] [string] $ThirdPartyCache,
+           [Parameter(Mandatory = $true)] [string] $VSSetupEnvScriptPath)
+    
+    Write-Host "Sourcing VS environment variables"
+    Invoke-BatchFile "$VSSetupEnvScriptPath"
+    
+    Write-Host "Copying common third-party dependencies"
+    New-Item -ItemType Directory .\third_party
+    Copy-Item -Recurse "$ThirdPartyCache\common\*" third_party\
+
+    Copy-Item tools\build\SConstruct .\
+}
+
+function Sign-MSI {
```

Comment:
Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)